client/java/src/main/java/io/openlineage/client/transports/HttpTransport.java

```diff
@@ -68,28 +70,34 @@ public HttpTransport(
     super(Type.HTTP);
     this.http = httpClient;
     try {
-      URI configUri = httpConfig.getUrl();
-      if (configUri.getPath() != null && !configUri.getPath().equals("")) {
-        if (httpConfig.getEndpoint() != null && !httpConfig.getEndpoint().equals("")) {
-          throw new OpenLineageClientException("You can't pass both uri and endpoint parameters.");
-        }
-        this.uri = httpConfig.getUrl();
-      } else {
-        String endpoint =
-            httpConfig.getEndpoint() != null && !httpConfig.getEndpoint().equals("")
-                ? httpConfig.getEndpoint()
-                : API_V1 + "/lineage";
-        this.uri =
-            new URIBuilder(httpConfig.getUrl())
-                .setPath(httpConfig.getUrl().getPath() + endpoint)
-                .build();
-      }
+      this.uri = getUri(httpConfig);
     } catch (URISyntaxException e) {
       throw new OpenLineageClientException(e);
     }
     this.tokenProvider = httpConfig.getAuth();
   }
 
+  private URI getUri(@NotNull HttpConfig httpConfig) throws URISyntaxException {
+    URI url = httpConfig.getUrl();
+    URIBuilder builder = new URIBuilder(url);
+    if (StringUtils.isNotBlank(url.getPath())) {
```

Comment:
could `&&` replace `if` nesting?

Reply:
Not sure what you mean. I see 👍 under my comment but code was not changed.

Reply:
it means it's a good idea, but for some reason I haven't implemented it... :/

Reply:
I stand corrected, i thought we could do it here but we can't because it would change the `else` condition